src/psyclone/tests/psyir/nodes/node_test.py

```diff
@@ -663,6 +663,18 @@ def test_node_ancestor_shared_with(fortran_reader):
     assert (assignment.ancestor(BinaryOperation, shared_with=one_lit)
             is None)
 
+    code2 = '''Subroutine test2()
+    integer :: x
+    x = 1 + 2
+    End Subroutine test2
+    '''
+    psyir2 = fortran_reader.psyir_from_source(code2)
+    assignment2 = psyir2.children[0].children[0]
+
+    assert one_lit.ancestor(Node, shared_with=assignment2) is None
+    assert one_lit.ancestor(Node, shared_with=assignment2,
```

Comment:
Comment please, e.g. "Test when supplied limit is *not* an ancestor of the original node (one_lit)"